earth_enterprise/rpms/buildSrc/src/main/groovy/com/tsciences/os/Platform.groovy

```diff
@@ -0,0 +1,64 @@
+package com.tsciences.os
+
+class Platform {
+    static private String osId = null;
+    static private String osVersionId = null;
+
+    static def obtanOsAndVersionIds() {
+        osId = null
+        osVersionId = null
+
+        if (new File('/etc/os-release').exists()) {
+            new File('/etc/os-release').readLines().each {
+                if (it.startsWith('ID=')) {
+                    osId = it.substring('ID='.length())
+                    if (osId.startsWith('"') && osId.endsWith('"')) {
+                        osId = osId.substring(1, osId.length() - 1)
+                    }
+                } else if (it.startsWith('VERSION_ID=')) {
+                    osVersionId = it.substring('VERSION_ID='.length())
+                    if (osVersionId.startsWith('"') && osVersionId.endsWith('"')) {
+                        osVersionId = osVersionId.substring(1, osVersionId.length() - 1)
+                    }
+                    // Only keep the major version number:
+                    osVersionId = osVersionId.find(/[0-9]+/)
+                }
+            }
+        } else if (new File('/etc/system-release-cpe').exists()) {
+            def osReleaseComponents = new File('/etc/system-release-cpe').text.split(':')
+            // The extract only the number from version strings like '6server':
+            osId = osReleaseComponents[2]
+            osVersionId = osReleaseComponents[4].find(/[0-9]+/)
+        }
+    }
```

Comment:
minor note: of course if we have neither, I presume osVersionId and osId remain null, which means getOsVersionId() & getOsId() will check again later in vein on the next rpm.  My suggestion, have it default to something like "unknown" and "0" if there is no result.

Reply:
If they're `null`, they'll expand to the `null` string, which somewhat like `unknown`, or `0`.  However, that can only happen on unsupported platforms where we don't have a defined build behavior.